src/main/java/org/opensearch/knn/index/codec/transfer/VectorTransferByte.java

```diff
@@ -34,11 +34,12 @@ public void init(final long totalLiveDocs) {
     public void transfer(final BytesRef bytesRef) {
         dimension = bytesRef.length * 8;
         if (vectorsPerTransfer == Integer.MIN_VALUE) {
-            // if vectorsStreamingMemoryLimit is 100 bytes and we have 50 vectors with length of 5, then per
-            // transfer we have to send 100/5 => 20 vectors.
             vectorsPerTransfer = vectorsStreamingMemoryLimit / bytesRef.length;
-            // If vectorsPerTransfer comes out to be 0, then we set number of vectors per transfer to 1, to ensure that
-            // we are sending minimum number of vectors.
+            if (totalLiveDocs > 0) {
```

Comment:
Why change this in this PR?

Reply:
The logic for calculating vectorsPerTransfer was incorrect when I first made this change. One of the new test cases also was not passing unless I limited vectorsPerTransfer to the number of documents, but this might be due to something else. Currently looking into this.

Reply:
Changed back